### ISSUE TITLE
Invalid call in canvasSwapChain!.ResizeBuffers() (0x887A0001)

### ISSUE THREAD

**Issue by KirillOsenkov** [Sergio0694/Win2DSample#1]
I'm seeing an exception at startup:

System.Runtime.InteropServices.COMException (0x887A0001): The application made a call that is invalid. Either the parameters of the call or the state of some object was incorrect. Enable the D3D debug layer in order to see details via debug messages. (0x887A0001)

![image](https://user-images.githubusercontent.com/679326/232344731-5ab4ec76-e6bb-4dbe-abcb-7d5aeaaee8e0.png)

<HTML>
<BODY>
<TABLE><TR><TD>&nbsp;</TD><TD>WinRT.Runtime</TD><TD>ExceptionHelpers.&lt;ThrowExceptionForHR&gt;g__Throw|20_0 Line 82</TD></TR><TR><TD>&nbsp;</TD><TD>WinRT.Runtime</TD><TD>ExceptionHelpers.ThrowExceptionForHR Line 85</TD></TR><TR><TD>&nbsp;</TD><TD>Microsoft.Graphics.Canvas.Interop</TD><TD>ICanvasSwapChainMethods.ResizeBuffers</TD></TR><TR><TD>&nbsp;</TD><TD>Microsoft.Graphics.Canvas.Interop</TD><TD>CanvasSwapChain.ResizeBuffers</TD></TR><TR><TD>&nbsp;</TD><TD>Win2DRenderer</TD><TD>Win32Application.OnUpdate Line 172</TD></TR><TR><TD>&nbsp;</TD><TD>Win2DRenderer</TD><TD>Win32ApplicationRunner.Run Line 121</TD></TR></TABLE>

</BODY>
</HTML>

Any quick idea what might be going wrong?

Thanks!

**Comment by Sergio0694** [comment]
Sorry, I'm using an internal Win2D build that has the necessary bits to make this work (you'll notice the package I'm using here isn't available in the public feed). For that resize issue specifically, the build is from a PR I did today, which isn't merged yet. Hopefully we should be able to publish everything this week. In the meantime, please feel free to ping me on Teams and I can share the internal package so you can try that locally if you want, that's what I've been doing as well 🙂

Out of curiosity - how come you're looking into this? 👀

**Comment by KirillOsenkov** [comment]
just curious :) and no rush. Just saw that you made this repo on GitHub and decided to check it out. Sounds like fun!

Thanks!

**Comment by Sergio0694** [comment]
Oh, cool! ahah
Yeah decided to put together a minimal sample to demonstrate how to use WinRT APIs for drawing from a plain Win32 app, as I've seen lots of people ask how to do that. Then we also went into a rabbit hole in the C# Discord server to make this run with NativeAOT, and figured I might as well open source the whole thing in case it could be useful for others too 😄

I'll let you know when the new Win2D bits are available and I've updated the sample, if you want to try it out then!
Cheers!
